implement post create course

teodorstrut opened the pull request:
I have added a method for creating a course. At the moment it only needs the `name` and `description` to create one.

teodorstrut committed:
implement post create course

i have added a method for creating a course. at the moment it only needs the name and description to create

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/services/course/CourseService.java:
```diff
@@ -27,4 +26,7 @@ public CourseService(CourseRepository courseRepository) {
     }
+
+    @Override
+    public void addCourse(Course c){this.courseRepository.save(c);}
```
Can you make this transactional ? We don't want corrupted data inside the database

teodorvoicencu replied:
and name the variable c as course

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/dtos/courses/CourseConverter.java:
```diff
@@ -22,6 +24,9 @@ public Course createFrom(CourseDTO dto) {
         course.description = dto.description;
+        Date date = new Date();
+        course.setCreatedDate(date);
+        course.setModifiedDate(date);
```
JPA automatically persists the audit related columns so this is not needed

teodorstrut committed:
minor fixes

-updated add method to be transactional
-removed unnecessary Data assignent to new Course object

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/dtos/courses/CourseConverter.java:
```diff
@@ -4,6 +4,8 @@
 import ubb.courses.backend.models.Course;
 
+import java.util.Date;
+
```
Remove unused imports

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/dtos/courses/CourseDTO.java:
```diff
@@ -10,6 +11,7 @@
 @EqualsAndHashCode(callSuper = true)
 public class CourseDTO extends BaseDTO {
     @NotNull
+    @Length(min = 4, max = 32)
```
Please use `@Size`

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/services/course/CourseService.java:
```diff
@@ -2,14 +2,16 @@
-import java.util.Optional;
 
 @Service
+@EnableTransactionManagement
```
Please put this annotation in `ManagementConfig`

teodorstrut committed:
minor fixes

-moved enableTransactionalManagement to ManagementConfig
-renamed method variable

teodorstrut committed:
minor fix

changed from @length to @size
removed unused imports

teodorvoicencu reviewed (approved):
LGTM
